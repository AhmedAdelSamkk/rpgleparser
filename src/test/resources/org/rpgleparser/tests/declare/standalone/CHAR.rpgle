        DCL-S library CHAR(10);
        DCL-S libfilembr VARCHAR(33);
        DCL-S cmdparm VARCHAR(10:4);