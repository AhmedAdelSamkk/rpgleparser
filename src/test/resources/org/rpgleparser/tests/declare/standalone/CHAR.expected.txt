/*===TOKENS===*/
DS_Standalone            <DCL-S>
ID                       <library>
ID                       <CHAR>
'('                      <(>
NUMBER                   <10>
')'                      <)>
FREE_SEMI                <;>
DS_Standalone            <DCL-S>
ID                       <libfilembr>
ID                       <VARCHAR>
'('                      <(>
NUMBER                   <33>
')'                      <)>
FREE_SEMI                <;>
DS_Standalone            <DCL-S>
ID                       <cmdparm>
ID                       <VARCHAR>
'('                      <(>
NUMBER                   <10>
COLON                    <:>
NUMBER                   <4>
')'                      <)>
FREE_SEMI                <;>
/*===TREE===*/
(r
  (dspec
    DCL-S
    (identifier (free_identifier library))
    (datatype
      (identifier (free_identifier CHAR))
      (args ( (expression (number 10)) ))
    )
    ;
  )
  (dspec
    DCL-S
    (identifier (free_identifier libfilembr))
    (datatype
      (identifier (free_identifier VARCHAR))
      (args ( (expression (number 33)) ))
    )
    ;
  )
  (dspec
    DCL-S
    (identifier (free_identifier cmdparm))
    (datatype
      (identifier (free_identifier VARCHAR))
      (args ( (expression (number 10)) : (expression (number 4)) ))
    )
    ;
  )
)
/*======*/